### PR TITLE
Apply gRPC error details to update project api

### PR DESCRIPTION
**Pull request by DongjinS**
**What this PR does / why we need it**:

- [x] deliver information for error details to [where the error convert to status](https://github.com/yorkie-team/yorkie/blob/f6a4aa15ea4cf3a13ae1288e1f9b191ac536aee0/server/grpchelper/status.go#L37)
    - delivered via custom error
- [x] fulfill the error details with proper key-value set
- [ ] handling error details at yorkie-house

**Which issue(s) this PR fixes**:

Fixes #338 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: api/types/updatable_project_fields.go [hunk 1]
```diff
@@ -47,6 +48,27 @@ var (
 	nameRegex = regexp.MustCompile("^[a-z0-9\\-._~]+$")
 )
 
+// ErrorWithDetails is error for deliver details with error
+type ErrorWithDetails struct {
+	err     error
+	details interface{}
```
Comment (hackerwins) [inline api/types/updatable_project_fields.go:54]: Can we use generics here?

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution.
I left a question.

File: api/types/updatable_project_fields.go (hunk 1 above)
Comment (DongjinS) [reply api/types/updatable_project_fields.go]: Is there a dependency issue on protobuf in the picture above?

**Review by hackerwins — APPROVED** [review]
Overall very neat. thank you.

**Pull request merged**